### PR TITLE
Milestone 4-Fetch API:Module-3

### PR DESCRIPTION
**Pull request by Tobinchilongo**
#### What this `PR` does :

- In the previous project you developed the add book and remove book features directly in your application using Redux and React state. Now you should refactor your code to use the Bookstore API to add and remove books.
- 
- Read the [Bookstore API documentation](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) to learn how to use the API.
- Add redux-thunk to your project.
- Use [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) or [axios](https://axios-http.com/docs/intro) for making API calls.
- Fetch book list from Bookstore API on Books page load.
    - Add an Action Creator, that returns a function.
    - Fetch list of books.
    - Dispatch another action (add an action type) that will pass API response to reducer and updates the state.
- Refactor your add book and remove book features to persist your changes in the server.
    - Change your existing Action Creators, so they return functions.
    - Make changes by using API.
    - Dispatch an action (action type that existed before refactor) that will updates the state accordingly.
- Add book and remove book should work in the same way after the refactor.
- No styling is required.

### CODE REVIEW

File: src/redux/configureStore.js [hunk 1]
```diff
@@ -11,7 +12,7 @@ const reducer = combineReducers({
 
 const store = createStore(
   reducer,
-  applyMiddleware(logger),
+  applyMiddleware(logger, thunk),
```
Comment (rotimiazeez) [inline src/redux/configureStore.js:15]: - [ ] Please make sure to meet the following requirement 💪🏽

- [ ] All Bookstore API calls are done with fetch OR Axios.
Action Creator for fetching books should be created

- [ ] It should return a function.

- [ ] It should fetch the books list from API.

- [ ] It should dispatch a new action type that will pass the books list to reducer and updates state.


- [ ] Action creator for adding a book should be refactored:

- [ ] It should return a function.

- [ ] It should add a new book via API.

- [ ] It should dispatch already existing ADD_BOOK (name can be slightly different).


- [ ] Action creator for removing a book should be refactored:

- [ ] It should return a function.

- [ ] It should remove a book via API.

- [ ] It should dispatch already existing REMOVE_BOOK (name can be slightly different).


- [ ] The Add book and remove book features should work the same as before even after the refactoring to use the API has been done.

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi @Tobinchilongo 👋, 
## Good job so far! 👍
### Summary 👏
- No linter errors ✔
- Good commit message 👍
- Descriptive PR message😊
- Good Readme 🚀
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js [hunk 2]
```diff
@@ -1,26 +1,58 @@
...
+  });
+};
 
-export const removeBook = (id) => ({
-  type: REMOVE_BOOK,
-  id,
-});
+export const fetchBooks = () => async (dispatch) => api
+  .fetchBooks().then((data) => dispatch({ type: GET_BOOKS, payload: handleData(data) }));
+
```
Comment (micheaol) [inline src/redux/books/books.js:23]: Your `removebook` method is not working as expected. If I have three(3) books, when I click on `remove` for one of the books, `the remove button will remove all the books available`. Kindly fix that.

**Review by micheaol — CHANGES_REQUESTED** [review]
Hi @Tobinchilongo , 

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## TO HIGHLIGHT: 
- Your project is professional 💯 
- Your repo is professional 💯 
- Well structured files 💯 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/redux/books/books.js (hunk 2 above)
Comment (Tobinchilongo) [reply src/redux/books/books.js:23]: Alright, Thank you. Let me work on it

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @Tobinchilongo,** 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## **Invalid Code Review Request**

You have submitted a project, that _**is not compiling due to added unrelated files [battlecry/generators/duck](https://github.com/Tobinchilongo/bookstore/tree/develop/battlecry/generators/duck)  and also the vscode folder**_

lllll 


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by FrederickMih — APPROVED** [review]
# Status: Approved :tada: :tada: 

Hi @Tobinchilongo , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — APPROVED** [review]
# Status: Approved :tada: :tada: 

Hi @Tobinchilongo , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**